src/Savegame/Country.cpp

```diff
@@ -251,6 +215,77 @@ void Country::newMonth(int xcomTotal, int alienTotal, int pactScore, int average
 		_activityXcom.erase(_activityXcom.begin());
 	if (_funding.size() > 12)
 		_funding.erase(_funding.begin());
+
+	return fundingChange;
+}
+
+/**
+ * Pure helper method to calculate country changes on new month.
+ * 
+ * There are five possible outcomes:
+ * - in a pact or formed a pact. No funding.
+ * - left a pact when last month's funding was 0. Country satisfied. Recieve average funding.
+ * - xcom score exceeds alien score by 30 points, and a random roll is succesful. Country happy. Increase funding.
+ * - a random roll exceeds xcom score. Country unhappy. Decrease funding.
+ * - None of the above. Country satisfied. No change to funding.
+ * @return change to funding and new satisfaction levels.
+*/
+[[nodiscard]] std::pair<int, Country::Satisfaction> Country::CalculateChanges(const OpenXcom::SavedGame& save, int xcomScore, int alienScore) const
+{
+	// prepare values.
+	// county specific scores
+	int countryXcomScore = (xcomScore / 10) + _activityXcom.back();
+	int countryAlienScore = (alienScore / 20) + _activityAlien.back();
+
+	// funding can change by 5 to 20%, in 1k increments, but always by at least 1000.
+	int lastMonthFunding = _funding.back();
+	int fundingPercentChange = RNG::generate(5, 20);
+	int fundingChange = (lastMonthFunding / 1000 * fundingPercentChange / 100) * 1000;
+	fundingChange = std::max(fundingChange, 1000);
+
+	// forming a pact, or in a pact and not canceling this month. No funding.
+	if (_newPact || (_pact && !_cancelPact))
+	{
+		// since next month funding is equal to lastMonthFunding + fundingChange, this will result in 0 funding next month.
+		// if funding was already 0, it will still be 0.
+		return { -lastMonthFunding, Satisfaction::ALIEN_PACT };
+	}
+
+	// canceling a pact and resetting country funding.
+	// note that if a pact is canceled but last month's funding was not 0, this branch is not taken.
+	// This can probably only happen via scripts though, which should handle the issue themselves.
+	if (_cancelPact && lastMonthFunding <= 0)
+	{
+		// using funding appropriate to the country's rules is probably better, but this is the original logic.
+		int averageCountryFunding = save.getCountryFunding() / save.getCountries()->size() / 1000 * 1000;
```

Comment:
this is not equal to original logic, because it can change as the countries are processed. original logic calculated this number only once, at the beginning